### PR TITLE
better error reporting for channel update failure

### PR DESCRIPTION
**Pull request by Bjvanminnen**
I was not a member of the deploy-status room this morning, and when using the `dotd` script, it silently failed to update the room message.

This makes it so that we will throw when we fail.

### CODE REVIEW

File: lib/cdo/slack.rb [hunk 1]
```diff
@@ -83,8 +83,17 @@ def self.update_topic(channel_name, new_topic, use_channel_map = false)
...
       "&topic=#{new_topic}"
     )
+    result = JSON.parse(response.read)
+    error = result['error']
+    if error
+      if error == 'not_in_channel'
+        raise "Must join room #{channel_name} to be able to update topic"
+      else
+        raise "Failed to update topic, with error: #{result['error']}"
+      end
```
Comment (Bjvanminnen) [inline lib/cdo/slack.rb]: The generic error of `Failed to update topic, with error: not_in_channel` did not seem super understandable without context, so I decided to special case that error with something more descriptive, and use the generic case for other things we might run into.

File: bin/dotd [hunk 2]
```diff
@@ -170,16 +167,12 @@ def dotd_menu_prompt
 
         #{bold 'Q'}: #{underline 'Q'}uit.
 
-    Your Choice: 
+    Your Choice:#{' '}
```
Comment (Bjvanminnen) [inline bin/dotd:170]: My editor auto-trims trailing whitespace. I suspect this is true of others as well. This seems like a more effective solution.

File: lib/cdo/slack.rb [hunk 3]
```diff
@@ -147,6 +148,18 @@ def self.message(text, params={})
     end
   end
 
+  def self.join_room(name)
+    response = open(
+      'https://slack.com/api/channels.join'\
+      "?token=#{SLACK_TOKEN}"\
+      "&name=#{name}"\
```
Comment (islemaster) [inline lib/cdo/slack.rb:155]: Looks like an extra trailing slash here.

Comment (Bjvanminnen) [reply lib/cdo/slack.rb:155]: I was following a pattern I saw elsewhere in the file (i.e. line 51), though I see we're not consistent (line 84 does not have one).

**Pull request merged**